llms/mlx_lm/lora.py

```diff
@@ -1,122 +1,63 @@
 import argparse
 import json
 import math
+import re
 from pathlib import Path
+from types import SimpleNamespace
 
 import mlx.optimizers as optim
 import numpy as np
+import yaml
 from mlx.utils import tree_flatten
 
 from .tuner.trainer import TrainingArgs, TrainingCallback, evaluate, train
 from .tuner.utils import linear_to_lora_layers
 from .utils import load
 
+yaml_loader = yaml.SafeLoader
+yaml_loader.add_implicit_resolver(
```

Comment:
What is the purpose of this?

Reply:
See: [Numbers in scientific notation without dot are parsed as string](https://github.com/yaml/pyyaml/issues/173)